Yoast/javascript
# LIN-80 Create sentence parser

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds a sentence parser to the tree.

## Relevant technical choices:

- In this implementation, the linguistic tree is not automatically executed when building the tree, but only when calling the `getTree` method on a text container.
- Note that parsing sentences into words isn't included in this PR yet. This will be addressed in a separate issue.
- The new `SentenceTokenizer` is a copy of the `SentenceTokenizer` already in use, with all HTML-specific functionality removed.
- I've removed all HTML-specific specs from the new `parseTextSpec.js`. I've also removed one particular non-html specific spec. In the old parser `This is a sentence. <` gets parsed into two sentence, `This is a sentence.` and `<`. Since all functionality specifically handling `<` tags was removed together with the other HTML-specific functionality, this now parses into 1 sentence, namely `This is a sentence. <`. I'd argue that the previous behavior was more of an artifact than really expected, especially in the current context where a `<` can't be an opening tag for a new HTML block anymore. Therefore I deleted this spec. (Same goes for `This is a sentence. < This sentence begins with a smaller than sign.`)
- I have copied the unit tests, expanded them with the indices and removed all tests that operated on HTML input (since we strip HTML elements before splitting into sentences now).
- I upped the max allowed ESLINT warnings because `SentenceTokenizer` was copied, and this function had a high complexity to begin with. Once we only use the tree-specific sentence tokenizer, the old tokenizer can be deleted.
- The issue described in #402 isn't fixed in this implementation.

## Test instructions

This PR can be tested by following these steps:

* Add more text variations to `parseTextSpec` and make sure they get parsed into sentences with correct content and indices.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes #406
